### PR TITLE
Update pandas version and fix `_add_columns_to_data`

### PR DESCRIPTION
**Pull request by fealho**
CU-86b03bmue.

Update pandas and some dev versions to match SDV. It also fixes the `_add_columns_to_data` for the so that it works with SDV Enterprise.

### CODE REVIEW

File: rdt/transformers/base.py [hunk 1]
```diff
@@ -292,10 +292,10 @@ def _add_columns_to_data(data, transformed_data, transformed_names):
         Returns:
             ``pandas.DataFrame`` with the new columns added.
         """
-        if isinstance(transformed_data, (pd.Series, np.ndarray)):
-            transformed_data = pd.DataFrame(transformed_data, columns=transformed_names)
-
         if transformed_names:
+            if isinstance(transformed_data, (pd.Series, np.ndarray)):
```
Comment (fealho) [inline rdt/transformers/base.py:296]: This is the important change.

**Comment by sdv-team** [comment]
Task linked: [CU-86b03bmue RDT - Update to work with SDV Enterprise](https://app.clickup.com/t/86b03bmue)

File: Makefile [hunk 2]
```diff
@@ -87,7 +87,7 @@ lint: ## check style with flake8 and isort
 fix-lint: ## fix lint issues using autoflake, autopep8, and isort
 	find rdt tests -name '*.py' | xargs autoflake --in-place --remove-all-unused-imports --remove-unused-variables
 	autopep8 --in-place --recursive --aggressive rdt tests
-	isort --apply --atomic --recursive rdt tests
```
Comment (pvk-developer) [inline Makefile:90]: why removing the `--recursive` ?

Comment (fealho) [reply Makefile:90]: It's [deprecated](https://pycqa.github.io/isort/docs/upgrade_guides/5.0.0.html).

File: rdt/transformers/base.py [hunk 3]
```diff
@@ -292,10 +292,10 @@ def _add_columns_to_data(data, transformed_data, transformed_names):
         Returns:
             ``pandas.DataFrame`` with the new columns added.
         """
-        if isinstance(transformed_data, (pd.Series, np.ndarray)):
-            transformed_data = pd.DataFrame(transformed_data, columns=transformed_names)
```
Comment (fealho) [inline rdt/transformers/base.py:296]: In the latest version of pandas, this raises an error if `columns=[]`, while before it used to return an empty dataframe. This causes the `RandomLocationGenerator` to fail in SDV Enterprise, so it needs to be fixed.

**Review by frances-h — APPROVED** [review]

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**